src/config.rs

```diff
@@ -26,7 +28,7 @@ const LOCAL_SERVER_HOST: &str = "127.0.0.1:8000";
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct HaConfig {
-    pub host: String,
+    pub host: Option<String>,
```

Comment:
Not sure i get why this is optional?

Reply:
So I made adding the host a part of the setup process. So technically when we first go to parse the config file, the way we can tell if we need to add run the add the host logic is by checking if this is None or not. 

I agree its a bit weird because it for sure isn't optional for the app to run but not sure of another good mechanism to let us know the user hasn't setup a host yet.